homeassistant/components/mazda/__init__.py

```diff
@@ -102,12 +156,22 @@ async def async_update_data():
     # Setup components
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
+    # Register services
+    for service in SERVICES:
+        hass.services.async_register(DOMAIN, service, async_handle_service_call)
```

Comment:
Please add a schema to validate the input instead of validating in the service call handler itself.
To validate the `device_id`, add a validator function to the schema.